### DISCUSSION TITLE
end to end encryption if using cascade connections

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [SoftEtherVPN/SoftEtherVPN#1793] [General]
Assuming a client interface (vNIC) as one end and a server interface (vNIC) as another end, we can say a VPN is end-to-end encryption.

```
# end to end encryption
[ client tap0/tun0 ] =====> [ server tap0/tun0 ]
```

I am wondering if using cascade connections (no SecureNAT) that relay traffic to next hop can be called end-to-end encryption ?

```
# is this one end to end encrypted ?
[ client tap0/tun0 ] =====> [ server-1 (cascade) ] =====> [ server-2 tap0/tun0 ]
```

if using cascade connections is not end-to-end encrypted, is there any way to have end-to-end encryption using SoftEther VPN?

**Comment by domosekai** [comment]
First of all, using the term end-to-end in the context of VPN is misleading.
What you are defining is client-server encryption, and you want it to be still true using cascade.
The answer is no if the cascading server is doing TLS with the client. In short, only if the client sees the end server's certificate in TLS handshake it is client-server-encryption.
